Add mango-rc4 hanlder

dwedul-figure opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

Create a `mango-rc4` upgrade handler to run the module migrations.

This PR is to update the `release/v1.11.x` branch. The PR to make the similar change to `main` is #896.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

channa-figure reviewed (approved):
Tested the upgrade locally and the version map was set properly along with the bank module param.
```
provenanced q params subspace bank SendEnabled
key: SendEnabled
subspace: bank
value: "null"
```

dwedul-figure commented:
I've also verified this.

Locally, I started with `v1.10.0` and updated the genesis file to have a `foofoocoin` entry in the bank module's `Params.SendEnabled`. I then put in an upgrade proposal, voted on it, waited for the halt, switched to this branch, built it and ran it. The chain started up just fine.

Prior to the upgrade:
```console
> build/provenanced q upgrade module_versions bank
module_versions:
- name: bank
  version: "2"
> build/provenanced q params subspace bank SendEnabled
key: SendEnabled
subspace: bank
value: '[{"denom":"foofoocoin"}]'
```

After the upgrade:
```console
> build/provenanced q upgrade module_versions bank
module_versions:
- name: bank
  version: "3"
> build/provenanced q params subspace bank SendEnabled
key: SendEnabled
subspace: bank
value: "null"
> build/provenanced q bank send-enabled
pagination:
  next_key: null
  total: "0"
send_enabled:
- denom: foofoocoin
  enabled: false
```

So the upgrade and migration happened as expected.

egaxhaj reviewed (approved):
LGTM
